plugin/builtin/input/file/file.go

```diff
@@ -205,6 +206,15 @@ func (f *FileInput) Start() error {
 	return nil
 }
 
+func (f *FileInput) warnIfNoMatches(matches []string, firstCheck bool) {
```

Comment:
I think it's strange to have a function that effectively says it will warn if there are no matches, yet it takes an argument that subverts this expectation based on its value. It might be clearer to just include this conditional above. Otherwise, unless you know how this function operates, it almost looks like we're warning on every match check.